File: springboot-backend/src/main/java/com/example/student/loan/service/LenderService.java
package com.example.student.loan.service;

import com.example.student.loan.model.Lender;
import com.example.student.loan.model.LenderDTO;
import com.example.student.loan.repository.LenderRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class LenderService {
    private final LenderRepository lenderRepository;

    public LenderService(LenderRepository lenderRepository) {
        this.lenderRepository = lenderRepository;
    }

    public List<Lender> getAllLenders() {
        return lenderRepository.findAll();
    }

    public LenderDTO getLenderDTOById(Integer id) {
        Lender lender = lenderRepository.findById(id).orElse(null);
        if(lender == null) return null;
        return LenderDTO.createDTO(lender);
    }

    public Lender getLenderById(Integer id) {
        return lenderRepository.findById(id).orElse(null);
    }

    public Lender createLender(Lender lender) {
        return lenderRepository.save(lender);
    }

    public Lender updateLender(Integer id, Lender lender) {
        lender.setLenderID(id);
        return lenderRepository.save(lender);
    }

    public void deleteLender(Integer id) {
        lenderRepository.deleteById(id);
    }

}

File: react-frontend/src/services/LenderService.js
import axios from 'axios';

const API_URL = 'http://127.0.0.1:8080/lenders';

class LenderService{
    getLenders(){
        return axios.get(API_URL);
    }

    getLenderById(id){
        return axios.get(API_URL + '/' + id);
    }

    getLenderByUserId(id){
        return axios.get(API_URL + '/user/' + id);
    }

    createLender(lender){
        return axios.post(API_URL, lender);
    }

    updateLender(lender, id){
        return axios.put(API_URL + '/' + id, lender);
    }

    deleteLender(id){
        return axios.delete(API_URL + '/' + id);
    }
}

export default new LenderService();